opencti-platform/opencti-graphql/src/parser/bundle-creator.ts

```diff
@@ -2,6 +2,43 @@ import { v4 as uuidv4 } from 'uuid';
 import * as R from 'ramda';
 import type { StixBundle, StixObject } from '../types/stix-common';
 import { STIX_SPEC_VERSION } from '../database/stix';
+import { STIX_EXT_OCTI } from '../types/stix-extensions';
+
+/**
+ * Check if bundle object can be added to the current bundle or if a new bundle is required to use upsert feature.
+ * If there is the same stix id but with different content => it need to be in another bundle because
+ * Same stix id on the one bundle are removed from processing during worker split process (worker see them as duplicate).
+ * @param objectsToAdd
+ * @param bundles
+ */
+export const canAddObjectToBundle = (objectsToAdd: StixObject[], bundles: StixObject[]): boolean => {
+  let canAdd = true;
+  for (let i = 0; i < objectsToAdd.length; i += 1) {
+    const currentToCheck = objectsToAdd[i];
+    const existingObjectWithDifferentContent = bundles.find((item: StixObject) => {
+      if (item.id === currentToCheck.id && item.type === currentToCheck.type) {
+        const itemClone = JSON.parse(JSON.stringify(item));
```

Comment:
I don't why we do a stringify => parse to create a clone? Can't we simplify it by using structuredClone?

Reply:
I used structuredClone(item) when I needed to do a deep clone. It is used in the buildRestrictedEntity of middleware.js for exemple. Is the JSON.parse(JSON.stringify(item)) creating a deep clone or is it doing some mofidication to the original item in the process?

Reply:
I need the same entity except for 2 fields => so I'm cloning and then removing 2 fields (that's why I need a clone)

Reply:
thanks trying structuredClone